microsoft/testsuites/stress/stress_ng_suite.py

```diff
@@ -112,34 +134,220 @@ def _run_stress_ng_job(
+        """
+        for node_index, node in enumerate(nodes):
+            try:
+                log.debug(f"Processing node {node_index + 1}/{len(nodes)}: {node.name}")
+
+                # Create dedicated workspace for stress-ng jobs
+                remote_workspace = node.working_path / "stress_ng_jobs"
+                node.shell.mkdir(remote_workspace, exist_ok=True)
+
+                # Deploy job file to remote node
+                remote_job_file = remote_workspace / job_file_name
+                node.shell.copy(PurePath(job_file), remote_job_file)
+
+                # Launch stress-ng with the job file
+                stress_process = node.tools[StressNg].launch_job_async(
+                    str(remote_job_file),
+                    yaml=True,
+                )
+                stress_processes.append(stress_process)
+
+            except Exception as deployment_error:
+                log.error(
+                    f"Failed to start stress job on node {node_index + 1}: "
+                    f"{deployment_error}"
+                )
+                if getattr(node, "log", None):
+                    node.log.error(f"Failed to start stress job: {deployment_error}")
+                raise deployment_error
+
+    def _monitor_stress_execution(
+        self,
+        stress_processes: List[Process],
+        nodes: List[RemoteNode],
+        log: Logger,
+        job_file_name: str,
+    ) -> tuple[TestStatus, str]:
+        """
+        Monitor stress-ng execution and capture stress-ng info output.
+
+        Returns:
+            Tuple of (TestStatus, stress_ng_info_output)
+        """
+
+        failed_nodes = 0
+        node_outputs = []
+        exceptions_to_raise = []
+
+        # Wait for all processes and capture their output
+        for i, process in enumerate(stress_processes):
+            node_name = nodes[i].name if i < len(nodes) else f"node-{i + 1}"
+            try:
+                process.wait_result(timeout=self.TIME_OUT, expected_exit_code=0)
+                log.info(f"{node_name} completed successfully")
+
+                # Process YAML output if applicable
+                node_output = self._process_yaml_output(nodes[i], job_file_name, log)
+
+                node_outputs.append(node_output)
+
+            except Exception as e:
+                failed_nodes += 1
+                error_output = f"=== {node_name} ===\nERROR: {str(e)}"
+                node_outputs.append(error_output)
+                log.error(f"{node_name} failed: {e}")
+                # Store the exception to re-raise after collecting all outputs
+                exceptions_to_raise.append(e)
+
+        # Combine all node outputs, including node names for clarity
+        execution_summary = f"Job: {job_file_name}\n\n"
+        for i, node_output in enumerate(node_outputs):
+            node_name = nodes[i].name if i < len(nodes) else f"node-{i + 1}"
+            execution_summary += f"=== {node_name} ===\n{node_output}\n\n"
```

Comment:
It looks to me like you're putting a lot of information into this execution_summary that ends up as the test result message in the DB. I don't think so much text is very readable in the DB. We should probably print it to the testcase logs using the logger.

Reply:
We are not putting all the logs in DB we are using this function to process the needed info "_process_yaml_output"